appveyor.yml

```diff
@@ -36,24 +39,35 @@ cache:
 shallow_clone: true
 
 install:
-  # We want  protoc.exe on PATH.
-  - SET PATH=C:/%RIPPLED_DEPS_PATH%;%PATH%
-
   # Download dependencies if appveyor didn't restore them from the cache.
   # Use 7zip to unzip.
   - ps: |
-        if (-not(Test-Path 'C:/$env:RIPPLED_DEPS_PATH')) {
-            echo "Download from $env:RIPPLED_DEPS_URL"
-            Start-FileDownload "$env:RIPPLED_DEPS_URL"
-            7z x "$($env:RIPPLED_DEPS_PATH).zip" -oC:\ -y > $null
-            if ($LastExitCode -ne 0) { throw "7z failed" }
+        if (-not(Test-Path "C:/$env:RIPPLED_DEPS_PATH/openssl")) {
+            $files = @(
+              "$env:RIPPLED_BOOST",
+              "$env:RIPPLED_BOOST_STAGE",
+              "$env:RIPPLED_OPENSSL"
+              )
+            For ($i=0; $i -lt $files.Length; $i++) {
+                $file = $files[$i]
+                $url = "$env:RIPPLED_DEPS_BASE_URL/$file"
+                echo "Download $file from $url"
+                Start-FileDownload "$url"
+                7z x "$file" -o"C:\$env:RIPPLED_DEPS_PATH" -y > $null
+                if ($LastExitCode -ne 0) { throw "7z failed" }
+            }
+        }
+        else
+        {
+            "Dependencies are in cache"
+            ls "C:/$env:RIPPLED_DEPS_PATH"
         }
 
   # Newer DEPS include a versions file.
   # Dump it so we can verify correct behavior.
   - ps: |
-        if (Test-Path "C:/$env:RIPPLED_DEPS_PATH/versions.txt") {
-          cat "C:/$env:RIPPLED_DEPS_PATH/versions.txt"
+        if (Test-Path "C:/$env:RIPPLED_DEPS_PATH/version*.txt") {
```

Comment:
since we know the version of deps file we're using, can't we safely assume this file exists? If it doesn't exist, would that be a sign is something wrong?

Reply:
There's more than one file - one per package. I think I agree that not having any files would be a sign of a problem.